Origin/module5 (#8)
bambera74/DevOps-Course-Starter: README.md

Ensure that your .env file is in the same root folder as the Dockerfile and contains the relevant information for your instance of Trello, API key, token and board / list IDs.

## To build to docker container

Run the following commands from a machine with docker installed.

To build a prod container run the command below from the same location as the dockerfile :

docker build --target production --tag todoapp:prod .

To build a dev container run the command below from the same location as the dockerfile :

docker build --target development --tag todoapp:dev .

## To run the container

Run the following command from a machine with docker installed. 

The command must be run from the same location as the Dockerfile:

DEV : 'docker run --publish 5001:5000 --env-file .env todoapp'

> PROD : 'docker run --publish 5002:8000 --env-file .env todoapp'

Reviewer: I was caught out at first by the image names not quite matching what we'd built
```suggestion
DEV : 'docker run --publish 5001:5000 --env-file .env todoapp:dev'

PROD : 'docker run --publish 5002:8000 --env-file .env todoapp:prod'
```

Note: add the -d flag to run in detached mode

For Dev you can then browse the site using 'http://{hostname_url or localhost}:5001'

For Prod you can then browse the site using 'http://{hostname_url or localhost}:5002'

## To run a development container with a bind mount

You may want to take advantage of the way that flask allows for dynamic reloads in your development container. To do this, you can mount the files on your local machine into the container using the bind mount command:

$ docker run --env-file ./.env -p 5100:80 --mount "type=bind,source=$(pwd)/todo_app,target=/app/todo_app" todo-app:dev

> i.e. docker run --publish 5001:5000 --env-file .env --mount "type=bind,source=/Users/ashleybamber/DevOps/devops-course-starter/todo_app,target=/opt/todo_app" todoapp:dev

Reviewer: It took me a few runs to work out why these two are different other than the change to the path (which shouldn't be necessary, as `$(pwd)` exists to give us a path-independent approach to the command) - it would be good to just offer a single command!
```suggestion
$ docker run --env-file ./.env -p 5001:500 --mount "type=bind,source=$(pwd)/todo_app,target=/opt/todo_app" todoapp:dev
```

## To exit the container running

> First run the 'docker container list' coomand to identify the unipque name of your running container.

Reviewer: Couple of typos here - minor but worth keeping an eye out for; your docs are a significant tool in the fight to keep people able to understand a codebase
```suggestion
First run the 'docker container list' command to identify the unique name of your running container.
```
